resources/js/Components/Modal/EventRegistrationModal.tsx

```diff
@@ -0,0 +1,157 @@
+import {useForm} from '@inertiajs/inertia-react'
+import {Backdrop, Box, Fade, Grid, Modal, Typography} from '@mui/material'
+import {Button, ErrorText, Select} from 'Components'
+import {useEffect} from 'react'
+import {useTranslation} from 'react-i18next'
+import styled from 'styled-components'
+import {CharacterData} from 'Types/character-data'
+import {EventData} from 'Types/event-data'
+import {SessionData} from 'Types/session-data'
+import route from 'ziggy-js'
+
+const StyledBox = styled(Box)`
```

Comment:
@pascwhale let's refactor this using the extracted `<Modal />` component to avoid duplication